cni-plugins/sriov-passthrough-cni/README.md

```diff
@@ -68,3 +68,12 @@ requests:
 limits:
   prow/sriov: "2"
 ```
+
+# Reserved system SR-IOV PFs
+Some nodes external network interfaces have SR-IOV capabilities, including the cluster network interface.
+The CNI may allocate such interfaces and cause network disconnection to the node.
+
+In order to prevent from the CNI to allocate such interfaces, set the `SYSTEM_RESERVED_PFS` with the interfaces that CNI should not allocate in a comma seperated form:
+```bash
+SYSTEM_RESERVED_PFS="eno1,eno2,eno3"
```

Comment:
We can consider a configmap that is mounted as a file
then we don't need to change the image
and can have empty config map where not needed (as the mount would fail otherwise)

We can also consider using MAC address in case we see that naming aren't good enough
due to ungraceful detach, that cause nics to be renamed - but we can start without it, and act only if
it happens.

Reply:
so we need to remove the `SYSTEM_RESERVED_PFS=""` isnt it ?

Reply:
> > We can consider a configmap that is mounted as a file then we don't need to change the image and can have empty config map where not needed (as the mount would fail otherwise)
> 
> We achieve the same with the env var, I prefer to start with an env var and if we see that a config-map is necessary we can change it later.
> 
> > Each machine will have its own configmap, with the same image.
> 
> Since the CNI is deployed using a daemonSet, the same config map will be mounted to all CNI pods, similar to the env var.

The configmap can be different on each machine in case needed,
but as long as we don't need we can start with env var.
Note that the old cluster will not support the repo, it is just need to be known.
 
> 
> > We can also consider using MAC address in case we see that naming aren't good enough due to ungraceful detach, that cause nics to be renamed - but we can start without it, and act only if it happens. The MAC can also use a configmap in case needed, so it will be the easiest way to add a file to the CNI pod.
> 
> Note that the interfaces specified in the env var will not be allocated by the CNI, and their names are note prone to be renamed.

Of course, I meant the real ones, they will be detached and might change their name into a prefix that is bad,
but if you use a full name it won't happen, unless a real one comes and goes with different name (due to bug or so) and then it won't be filtered, we can start without and handle of course only if it happens.

Reply:
> The configmap can be different on each machine in case needed,
but as long as we don't need we can start with env var.
Note that the old cluster will not support the repo, it is just need to be known.

IIUC it means that there will be config-map object per each node, all config-maps should be mounted to the CNI pod, and the CNI should check against all files before allocation.
I think it's harder to maintain.

> Of course, I meant the real ones, they will be detached and might change their name into a prefix that is bad,
but if you use a full name it won't happen unless a real one comes and goes with different name (due to bug or so) and then it won't be filtered, we can start without and handle of course only if it happens.

If you meant for interfaces that will get allocated, in case their name change, the  CNI will still not allocate the reserved interfaces (which are specified in the env var).

Reply:
> > The configmap can be different on each machine in case needed,
> > but as long as we don't need we can start with env var.
> > Note that the old cluster will not support the repo, it is just need to be known.
> 
> IIUC it means that there will be config-map object per each node, all config-maps should be mounted to the CNI pod, and the CNI should check against all files before allocation. I think it's harder to maintain.
> 

You will have configmap per SR-IOV BM.
In case each BM need other list it won't able to use correct solution.
The prow job that deploy the cluster will know to set those configmaps.
The configmap will be mounted as a file, and then the script will read it to build the black list.
If the current static list would be enough there is no need to.

> > Of course, I meant the real ones, they will be detached and might change their name into a prefix that is bad,
> > but if you use a full name it won't happen unless a real one comes and goes with different name (due to bug or so) and then it won't be filtered, we can start without and handle of course only if it happens.
> 
> If you meant for interfaces that will get allocated, in case their name change, the CNI will still not allocate the reserved interfaces (which are specified in the env var).

This what I wrote "but if you use a full name it won't happen..."
beside the little corner case, which is just theoretical atm, so we can ignore it unless it happens.